src/internal/direction.ts

```diff
@@ -1,8 +1,20 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-export function isRtl(element: HTMLElement | SVGElement) {
-  return getComputedStyle(element).direction === 'rtl';
+export type IsRtl = boolean;
+export type ElementWithDirection = Document | HTMLElement | SVGElement;
+
+export function getIsRtl(element: ElementWithDirection) {
+  if (element instanceof Document) {
+    return document.documentElement.dir === 'rtl';
+  } else {
+    return getComputedStyle(element).direction === 'rtl';
+  }
+}
+
+export function getOffsetInlineStart(element: HTMLElement) {
+  const offsetParentWidth = element.offsetParent?.clientWidth ?? 0;
+  return getIsRtl(element) ? offsetParentWidth - element.clientWidth - element.offsetLeft : element.offsetLeft;
```

Comment:
Why is this `clientWidth` and not `offsetWidth`?

Reply:
I can use offsetWidth for the element but the offsetParent is typed as Element and will only allow for clientWidth.

Reply:
These values are not equivalent, they have different use-cases.

Which one do we need to make it work correctly?

--

P.S. Since this function is used only once and this whole usage is a hack, can you think of a better way to refactor this whole piece in RTL-friendly way?

Reply:
`offsetWidth` is the more accurate measurement because it is inclusive of borders and margins. In the `mainOffsetLeft` use case they are computing the same, but `offsetWidth` will be more resilient when used in max width computations. As far alternatives, I have the same issue with `offsetLeft` as I do with `getBoundingClientRect` in that they are absolute values from the physical x position regardless of document direction. If you have other ideas please share them.

Reply:
Your bet, but I would not introduce a reusable function for that and just do that calculation inline.

Reusable function tells other developers "here is a working implementation, feel free to use without worrying about internals". Can you say this about the function?

Reply:
I have at least one more occurrence that I know of which will require the same computation. I'm going to keep the utility function for now.

Reply:
The question if this function is really reusable by anyone without deep understanding of the matter, still stays